.github/workflows/ci.yml

```diff
@@ -0,0 +1,52 @@
+name: CI
+
+on:
+  pull_request:
+    branches: [$default-branches]
+  push:
+    branches: [$default-branch]
+
+jobs:
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    services:
+      postgres:
+        images: postgres
+        env:
+          POSTGRES_DB: upsert_test
+          POSTGRES_PASSWORD: postgres
+          POSTGRES_USER: postgres
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+        ports:
+          - 5432:5432
+
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby-version: [2.5.x, 2.6.x, 2.7.x]
```

Comment:
NOTE: I've removed 2.4 as it's EOL https://www.ruby-lang.org/en/news/2020/04/05/support-of-ruby-2-4-has-ended/

Reply:
Thanks for the inspiration!

In order to make it formal, and _perhaps_ easier to loate, I'll make a small PR with a commit taking that version out, before we take this in.

(While doing that, I noticed ruby2.3 packages being used in Dockerfile. So, some spring cleaning was needed.